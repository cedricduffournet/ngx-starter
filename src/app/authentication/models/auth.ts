export interface AuthToken {
  access_token: string;
  expires_in: number;
  token_type: string;
  refresh_token: string;
}

export interface Credentials {
  login: string;
  password: string;
}

export interface ResetPassword {
  plainPassword: {
    first: string;
    second: string;
  };
}

export interface Register extends ResetPassword {
  civility: number;
  firstname: string;
  lastname: string;
  email: string;
  confirmationUrl: string;
}

export interface SendResetPassword {
  username: string;
}
