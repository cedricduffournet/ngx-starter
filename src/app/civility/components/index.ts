export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
