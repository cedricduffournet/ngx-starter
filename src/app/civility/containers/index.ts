export * from './civility-add-modal/civility-add-modal.component';
export * from './civility-update-modal/civility-update-modal.component';
export * from './civility-delete-modal/civility-delete-modal.component';
export * from './civility-list-view/civility-list-view.component';
