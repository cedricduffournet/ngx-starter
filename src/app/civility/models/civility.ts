import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
