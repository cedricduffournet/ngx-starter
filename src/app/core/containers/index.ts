export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
