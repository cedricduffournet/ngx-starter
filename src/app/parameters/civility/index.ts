export * from './parameter-civility.component';
