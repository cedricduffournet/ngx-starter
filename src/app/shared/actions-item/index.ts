export * from './actions-item.module';
