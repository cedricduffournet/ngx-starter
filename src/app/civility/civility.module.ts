import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityEffects } from '@app/civility/state/effects';
import {
  CivilityAddComponent,
  CivilityUpdateComponent,
  CivilityDeleteComponent,
  CivilityFormComponent,
  CivilityItemComponent,
  CivilityItemsComponent
} from '@app/civility/components';
import {
  CivilityAddModalComponent,
  CivilityUpdateModalComponent,
  CivilityDeleteModalComponent,
  CivilityListViewComponent
} from '@app/civility/containers';
import { reducers } from '@app/civility/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([CivilityEffects]),
    StoreModule.forFeature('civilities', reducers)
  ],
  declarations: [
    CivilityListViewComponent,
    CivilityAddComponent,
    CivilityUpdateComponent,
    CivilityDeleteComponent,
    CivilityFormComponent,
    CivilityItemComponent,
    CivilityItemsComponent,
    CivilityAddModalComponent,
    CivilityUpdateModalComponent,
    CivilityDeleteModalComponent
  ],
  entryComponents: [
    CivilityUpdateModalComponent,
    CivilityAddModalComponent,
    CivilityDeleteModalComponent
  ],
  exports: [CivilityListViewComponent]
})
export class CivilityModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
