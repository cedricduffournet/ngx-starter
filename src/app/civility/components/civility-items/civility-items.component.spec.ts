import { TestBed, ComponentFixture, async } from '@angular/core/testing';

import { TranslateModule } from '@ngx-translate/core';

import { ActionsItemsModule } from '@app/shared/actions-item';
import { ButtonModule } from '@app/shared/button';
import {
  CivilityItemsComponent,
  CivilityItemComponent
} from '@app/civility/components';
import { Civility } from '@app/civility/models/civility';

describe('CivilityItems', () => {
  let fixture: ComponentFixture<CivilityItemsComponent>;
  let component: CivilityItemsComponent;
  const civilities: Civility[] = [
    {
      id: 1,
      code: 'testcode1',
      name: 'testname1'
    },
    {
      id: 2,
      code: 'testcode2',
      name: 'testname2'
    }
  ];

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot(), ActionsItemsModule, ButtonModule],
      declarations: [CivilityItemsComponent, CivilityItemComponent]
    });
    fixture = TestBed.createComponent(CivilityItemsComponent);
    component = fixture.componentInstance;
    component.civilities = civilities;
    component.authorization = {
      create: false,
      update: false,
      delete: false
    };
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit delete event', () => {
    spyOn(component.delete, 'emit');
    component.onDelete(civilities[0].id);
    expect(component.delete.emit).toHaveBeenCalledWith(1);
  });

  it('should emit edit event', () => {
    spyOn(component.update, 'emit');
    component.onUpdate(civilities[1].id);
    expect(component.update.emit).toHaveBeenCalledWith(2);
  });

  it('should emit add event', () => {
    spyOn(component.add, 'emit');
    component.onAdd();
    expect(component.add.emit).toHaveBeenCalledWith('add');
  });

  it('should display add button', () => {
    component.authorization = {
      update: true,
      delete: true,
      create: true
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should not display add button', () => {
    component.authorization = {
      update: true,
      delete: true,
      create: false
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
