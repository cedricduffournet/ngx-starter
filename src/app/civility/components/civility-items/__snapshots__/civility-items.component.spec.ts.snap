// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivilityItems should be created 1`] = `
<app-civility-items
  add={[Function EventEmitter]}
  authorization={[Function Object]}
  civilities={[Function Array]}
  delete={[Function EventEmitter]}
  update={[Function EventEmitter]}
>
  <section
    class="bg-white border-left border-right border-bottom rounded"
  >
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover"
      >
        <thead>
          <tr>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               ABBREVIATION 
            </th>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               NAME 
            </th>
            <th
              class="align-middle text-right"
            >
               Actions 
            </th>
          </tr>
        </thead>
        <tbody>
          
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode1
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname1 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="false"
                  ng-reflect-with-edit="false"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode2
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname2 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="false"
                  ng-reflect-with-edit="false"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</app-civility-items>
`;

exports[`CivilityItems should display add button 1`] = `
<app-civility-items
  add={[Function EventEmitter]}
  authorization={[Function Object]}
  civilities={[Function Array]}
  delete={[Function EventEmitter]}
  update={[Function EventEmitter]}
>
  <section
    class="bg-white border-left border-right border-bottom rounded"
  >
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover"
      >
        <thead>
          <tr>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               ABBREVIATION 
            </th>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               NAME 
            </th>
            <th
              class="align-middle text-right"
            >
               Actions 
            </th>
          </tr>
        </thead>
        <tbody>
          
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode1
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname1 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="true"
                  ng-reflect-with-edit="true"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,pencil-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-pencil-alt fa-w-16"
                          data-icon="pencil-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                    
                    <button
                      class="btn btn-outline-danger"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,trash-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-trash-alt fa-w-14"
                          data-icon="trash-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode2
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname2 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="true"
                  ng-reflect-with-edit="true"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,pencil-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-pencil-alt fa-w-16"
                          data-icon="pencil-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                    
                    <button
                      class="btn btn-outline-danger"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,trash-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-trash-alt fa-w-14"
                          data-icon="trash-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section><app-button-circle-right
    label="+"
    ng-reflect-label="+"
  >
    <div>
      <button
        class="btn-circle btn-xl btn btn-absolute-corner-right btn-success"
        ng-reflect-klass="btn-circle btn-xl btn btn-abso"
        ng-reflect-ng-class="btn-success"
        type="button"
      >
         + 
      </button>
    </div>
  </app-button-circle-right>
</app-civility-items>
`;

exports[`CivilityItems should not display add button 1`] = `
<app-civility-items
  add={[Function EventEmitter]}
  authorization={[Function Object]}
  civilities={[Function Array]}
  delete={[Function EventEmitter]}
  update={[Function EventEmitter]}
>
  <section
    class="bg-white border-left border-right border-bottom rounded"
  >
    <div
      class="table-responsive"
    >
      <table
        class="table table-hover"
      >
        <thead>
          <tr>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               ABBREVIATION 
            </th>
            <th
              class="align-middle"
              ng-reflect-translate=""
              translate=""
            >
               NAME 
            </th>
            <th
              class="align-middle text-right"
            >
               Actions 
            </th>
          </tr>
        </thead>
        <tbody>
          
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode1
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname1 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="true"
                  ng-reflect-with-edit="true"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,pencil-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-pencil-alt fa-w-16"
                          data-icon="pencil-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                    
                    <button
                      class="btn btn-outline-danger"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,trash-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-trash-alt fa-w-14"
                          data-icon="trash-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
          <tr
            appcivilityitem=""
            class="item"
            ng-reflect-authorization="[object Object]"
            ng-reflect-civility="[object Object]"
          >
            <td
              class="align-middle"
            >
              testcode2
            </td>
            <td
              class="align-middle"
            >
              <a
                class="h5"
                href="#"
                title="UPDATE_ELEM"
              >
                 testname2 
              </a>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="float-right actions"
              >
                <app-actions-item
                  ng-reflect-with-delete="true"
                  ng-reflect-with-edit="true"
                >
                  <div
                    class="button-validation-action"
                  >
                    
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,pencil-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-pencil-alt fa-w-16"
                          data-icon="pencil-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                    
                    <button
                      class="btn btn-outline-danger"
                      type="button"
                    >
                      <fa-icon
                        class="ng-fa-icon"
                        ng-reflect-icon-prop="fas,trash-alt"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-trash-alt fa-w-14"
                          data-icon="trash-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                            fill="currentColor"
                          />
                        </svg>
                      </fa-icon>
                    </button>
                  </div>
                </app-actions-item>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</app-civility-items>
`;
